### PR TITLE
Update tlsio_openssl to send SNI on client hello

### PR DESCRIPTION
**Pull request by yunhaoling**
Update tlsio_openssl to send SNI on client hello

addressing the issue: https://github.com/Azure/azure-uamqp-python/issues/167

and code is copied (slight adjusted) from here: https://github.com/Azure/azure-c-shared-utility/pull/453

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
 
+1.2.11 (Unreleased)
++++++++++++++++++++
+
+- Updated tlsio_openssl module to send SNI when establishing tls connection (Thanks to milope).
+
+1.2.10 (Unreleased)
++++++++++++++++++++
+
+- Added parameter `shutdown_after_timeout` to `ReceiveClient` and `ReceiveClientAsync` which gives control over whether to shutdown receiver after timeout.
```
Comment (KieranBrantnerMagee) [inline HISTORY.rst:14]: dup?
(would remove in suggestion but can't do multi-line ><)

File: src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_openssl.c [hunk 2]
```diff
@@ -1045,6 +1047,17 @@ static int create_openssl_instance(TLS_IO_INSTANCE* tlsInstance)
                         log_ERR_get_error("Failed creating OpenSSL instance.");
                         result = __FAILURE__;
                     }
+                    else if (SSL_set_tlsext_host_name(tlsInstance->ssl, tlsInstance->hostname) != 1)
```
Comment (KieranBrantnerMagee) [inline src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_openssl.c:1050]: for my own education; this is our own patched version of the shared-utils dep?  we don't just bring over their changes wholesale?  (I can see why we might do that, just want to understand fully and not assume)

File: HISTORY.rst (hunk 1 above)
Comment (yunhaoling) [reply HISTORY.rst:14]: LOL, that's something I copied and forgot to delete, thanks for pointing out ><

File: src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_openssl.c (hunk 2 above)
Comment (yunhaoling) [reply src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_openssl.c:1050]: no, not our own. it's simply a copy paste of the fix from the shared-utils :P

**Review by annatisch — APPROVED** [review]
LGTM 👍

**Pull request merged**